server/src/main/java/org/opensearch/common/unit/Fuzziness.java

```diff
@@ -192,6 +195,9 @@ public static Fuzziness parse(XContentParser parser) throws IOException {
                     // Validate if the fuzziness value is formatted correctly as a numeric value.
                     try {
                         final float minimumSimilarity = Float.parseFloat(fuzziness);
+                        if (minimumSimilarity < 0) {
```

Comment:
Super nit:

```suggestion
                        if (minimumSimilarity < 0.0f) {
```

Also, could you please add checks for `Float.isInfinite` and `Float.isNaN`?

Reply:
Addressed changes in [d28d3e2](https://github.com/opensearch-project/OpenSearch/pull/7154/commits/d28d3e20e5ea69330b50904fa4f843a7431fdcc0), thanks for the suggestions